x-pack/plugins/ml/public/application/components/scatterplot_matrix/scatterplot_matrix_view.tsx

```diff
@@ -0,0 +1,318 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useMemo, useEffect, useState, FC } from 'react';
+
+// There is still an issue with Vega Lite's typings with the strict mode Kibana is using.
+// @ts-ignore
+import { compile } from 'vega-lite/build-es5/vega-lite';
+import { parse, View, Warn } from 'vega';
+import { Handler } from 'vega-tooltip';
+
+import {
+  htmlIdGenerator,
+  EuiComboBox,
+  EuiComboBoxOptionOption,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiSelect,
+  EuiSwitch,
+} from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+import type { SearchResponse7 } from '../../../../common/types/es_client';
+
+import { useMlApiContext } from '../../contexts/kibana';
+
+import { getProcessedFields } from '../data_grid';
+import { useCurrentEuiTheme } from '../color_range_legend';
+
+import { ScatterplotMatrixLoading } from './scatterplot_matrix_loading';
+
+import {
+  getScatterplotMatrixVegaLiteSpec,
+  LegendType,
+  OUTLIER_SCORE_FIELD,
+} from './scatterplot_matrix_vega_lite_spec';
+
+import './scatterplot_matrix_view.scss';
+
+const SCATTERPLOT_MATRIX_DEFAULT_FIELDS = 4;
+const SCATTERPLOT_MATRIX_DEFAULT_FETCH_SIZE = 1000;
+const SCATTERPLOT_MATRIX_DEFAULT_FETCH_MIN_SIZE = 1;
+const SCATTERPLOT_MATRIX_DEFAULT_FETCH_MAX_SIZE = 10000;
+
+const TOGGLE_ON = i18n.translate('xpack.ml.splom.toggleOn', {
+  defaultMessage: 'On',
+});
+const TOGGLE_OFF = i18n.translate('xpack.ml.splom.toggleOff', {
+  defaultMessage: 'Off',
+});
+
+const sampleSizeOptions = [100, 1000, 10000].map((d) => ({ value: d, text: '' + d }));
+
+export interface ScatterplotMatrixViewProps {
+  fields: string[];
+  index: string;
+  resultsField?: string;
+  color?: string;
+  legendType?: LegendType;
+}
+
+export const ScatterplotMatrixView: FC<ScatterplotMatrixViewProps> = ({
```

Comment:
Is the query set in the query bar used when loading the data for the scatterplot matrix? It doesn't look like it is:

![image](https://user-images.githubusercontent.com/7405507/105865658-0b82b200-5feb-11eb-9072-35eed3276790.png)